### PR TITLE
Metadata, Docs and Template for IHC Assay

### PR DESCRIPTION
**Pull request by roshni-b**

### CODE REVIEW

File: cidc_schemas/schemas/assays/components/imaging/ihc_entry.json [hunk 1]
```diff
@@ -0,0 +1,53 @@
+{
```
Comment (jim-bo) [inline cidc_schemas/schemas/assays/components/imaging/ihc_entry.json]: So for percentages I would make the values and integer between 0-100: https://json-schema.org/understanding-json-schema/reference/numeric.html

For the "intensity" measurements I would make them an integer as well and limit it to 0,1,2,3 for now.

File: cidc_schemas/schemas/templates/metadata/ihc_template.json [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+                  "type_ref": "sample.json#properties/cimac_id"
+                },
+                "slide number": {
+                  "merge_pointer": "0/slide_number",
+                  "type_ref": "aliquot.json#properties/slide_number"
+                },
+                "IHC image path": {
+                  "merge_pointer": "0/files/ihc_image_path",
+                  "is_artifact" : 1,
+                  "gcs_uri_format": "{protocol identifier}/ihc/ihcimage",
```
Comment (jim-bo) [inline cidc_schemas/schemas/templates/metadata/ihc_template.json]: This should have the cimac_id and slide number to make it unique, also it should have the correct file suffix. I would set this to be something like:
"{protocol identifier}/{cimac id}/{slide number}/ihc/ihc.tiff"

File: cidc_schemas/schemas/templates/metadata/ihc_template.json [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+                  "type_ref": "assays/components/local_file.json#properties/file_path"
+                },
+                "H and E image path": {
+                  "merge_pointer": "0/files/he_image_path",
+                  "is_artifact" : 1,
+                  "gcs_uri_format": "{protocol identifier}/ihc/heimage",
```
Comment (jim-bo) [inline cidc_schemas/schemas/templates/metadata/ihc_template.json]: Similar to the above comment I would do something like this:
"{protocol identifier}/{cimac id}/{slide number}/ihc/hande.tiff"

File: cidc_schemas/schemas/assays/components/imaging/ihc_input.json [hunk 4]
```diff
@@ -0,0 +1,14 @@
+{
```
Comment (jim-bo) [inline cidc_schemas/schemas/assays/components/imaging/ihc_input.json]: This doesn't seem to matchup with what you've put in the ihc_template.json...
For "IHC image path" you set the merge pointer to be "0/files/ihc_image_path" and in "ihc_entry.json" you have "files" -> "ihc_input.json". This means that ihc_input.json should have property named "ihc_image_path".

Similar comment for the H and E image.

File: cidc_schemas/schemas/assays/components/imaging/ihc_entry.json (hunk 1 above)
Comment (roshni-b) [reply cidc_schemas/schemas/assays/components/imaging/ihc_entry.json]: Alright. Do you think the "Average intensity" should allow floating point numbers? In that case  I'll set that type: `number`, and the rest to type: 'integer'

**Comment by curlup** [comment]
> 2. Next I would create a new unit test in "test_assays.py" where you mock-up a json object that should validate against ihc_assay.json

Or you could do this as a `.json` of an CT with such assay in 'tests/data/clinicaltrial_examples/` dir - it will automatically check against CT and assay schema.

File: cidc_schemas/schemas/assays/components/imaging/ihc_entry.json (hunk 1 above)
Comment (jim-bo) [reply cidc_schemas/schemas/assays/components/imaging/ihc_entry.json]: Number works.

Comment (jim-bo) [reply cidc_schemas/schemas/assays/components/imaging/ihc_entry.json]: Number works, but I would specify the allowed range as well (0-100) and (0-3) ...inclusive...for the different types.

File: tests/test_assays.py [hunk 5]
```diff
@@ -254,6 +254,49 @@ def test_cytof():
     validator.validate(obj)
 
 
+def test_ihc():
```
Comment (jim-bo) [inline tests/test_assays.py]: I would first make all fields required in the data model, then make sure this example has a true example of what the example JSON looks like for an IHC experiment.

**Pull request merged**